src/applications/disability-benefits/all-claims/tests/utils/utils.unit.spec.jsx

```diff
@@ -670,6 +743,47 @@ describe('isAnswering781aQuestions', () => {
     });
   });
 
+  describe('PTSD flow migration', () => {
+    const ptsdFormData = {
+      newDisabilities: [
+        {
+          condition: 'Ptsd personal trauma',
+        },
+      ],
+      'view:selectablePtsdTypes': {
+        'view:assaultPtsdType': true,
+      },
+      'view:upload781aChoice': 'answerQuestions',
+      secondaryIncident0: {
+        'view:uploadSources': true,
+      },
+    };
+
+    describe('when the modern 0781 flow is enabled', () => {
+      const ptsdFormDataModernFlowEnabled = {
+        syncModern0781Flow: true,
+        ...ptsdFormData,
```

Comment:
Similar question concerning conflicting `formData` based on `syncModern0781Flow` state.

Reply:
Yeah these are good call outs. So the purpose of rewriting all of these tests is to ensure that if for whatever reason all of the conditions exist for these legacy flow pages to display, the state of the new flow flipper will still prevent them from displaying. It's the most deterministic way I can think of to ensure the change doesn't leak any legacy pages when we don't want to display them.

I think the tradeoff here, like you mentioned, is we are testing things that shouldn't happen. So I dunno, given that would removing these tests be something you would want before approving?

Reply:
I suppose, provided that rationale, knowing that you can't simply test the presence of `syncModern0781Flow: true` determining the return value of a function, that other key values are influential, then the improbable form data being tested will provide extra assurance.